tests/test_operation.py

```diff
@@ -177,6 +177,44 @@ def test_operation_init(self, test_class, monkeypatch):
         if n == 0:
             return
 
+    def test_barrier(self):
+        r"Test that the barrier influences compilation"
+
+        def qfunc():
+            qml.Hadamard(wires=0)
+            qml.Barrier(wires=0)
```

Comment:
Here and in the next test case too, it can be useful to test more circuits that may be considered edge cases:
* What if we have multiple barriers one after the other?
* What if we have operations in between barriers?
* What if we have barriers right at the beginning/end of the circuit?